components/org.wso2.carbon.identity.webfinger/src/test/java/org/wso2/carbon/identity/webfinger/builders/DefaultWebFingerRequestBuilderTest.java

```diff
@@ -0,0 +1,238 @@
+
+@PrepareForTest({HttpServletRequest.class, WebFingerServiceComponentHolder.class, RealmService.class,
+        TenantManager.class})
+public class DefaultWebFingerRequestBuilderTest extends PowerMockTestCase {
+    private DefaultWebFingerRequestBuilder defaultWebFingerRequestBuilder;
+    private String rel;
+    private String resource;
+    private String host;
+    private String tenant;
+    private String path;
+    private String scheme;
+    private int port;
+    private String acctResource;
+    private String invalidAcctResource;
+
+    @Mock
+    HttpServletRequest request;
+
+    @Mock
+    WebFingerServiceComponentHolder webFingerServiceComponentHolder;
+
+    @Mock
+    RealmService realmService;
+
+    @Mock
+    TenantManager tenantManager;
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+        defaultWebFingerRequestBuilder = new DefaultWebFingerRequestBuilder();
+        rel = "http://openid.net/specs/connect/1.0/issuer";
+        resource = "https://oidc.testdomain.wso2.org:9443/.well-known/webfinger";
+        host = "oidc.testdomain.wso2.org";
+        tenant = "carbon.super";
+        path = "/.well-known/webfinger";
+        acctResource = "acct:joe@example.com";
+        invalidAcctResource = "acct:joe.example.com";
+        scheme = "https";
+        port = 9443;
+
+        mockStatic(WebFingerServiceComponentHolder.class);
+        when(WebFingerServiceComponentHolder.getRealmService()).thenReturn(realmService);
+        when(realmService.getTenantManager()).thenReturn(tenantManager);
+    }
+
+    @DataProvider(name = "BuildParameters")
+    public Object[][] buildScopeString() {
+        Vector<String> vector1 = new Vector<>();
+        vector1.add("value1");
+        vector1.add("value2");
+        Enumeration<String> params1 = vector1.elements();
+
+        Vector<String> vector2 = new Vector<>();
+        vector2.add("value1");
+        Enumeration<String> params2 = vector2.elements();
+
+        Vector<String> vector3 = new Vector<>();
+        vector3.add(WebFingerConstants.REL);
+        vector3.add("value2");
+        Enumeration<String> params3 = vector3.elements();
+
+        Vector<String> vector4 = new Vector<>();
+        vector3.add(WebFingerConstants.RESOURCE);
+        vector3.add("value2");
+        Enumeration<String> params4 = vector4.elements();
+        return new Object[][] {
+                {params1},
+                {params2},
+                {params3},
+                {params4}
+        };
+    }
+    //Test for exception "Bad webfinger request"
+    @Test(dataProvider = "BuildParameters")
+    public void testInvalidRequest(Enumeration<String> params) {
+        when(request.getParameterNames()).thenReturn(params);
+        try {
+            defaultWebFingerRequestBuilder.buildRequest(request);
+        } catch (WebFingerEndpointException e) {
+            assertEquals(e.getErrorCode(), WebFingerConstants.ERROR_CODE_INVALID_REQUEST);
```

Comment:
A better option would be to do the assert and rethrow it. And define it as an expected exception. Otherwise what if actually the exception is not occured ? Still the test will be successful